rustls/src/anchors.rs

```diff
@@ -82,8 +83,9 @@ impl RootCertStore {
     }
 
     /// Add a single DER-encoded certificate to the store.
-    pub fn add(&mut self, der: &key::Certificate) -> Result<(), webpki::Error> {
-        let ta = webpki::TrustAnchor::try_from_cert_der(&der.0)?;
+    pub fn add(&mut self, der: &key::Certificate) -> Result<(), Error> {
+        let ta = webpki::TrustAnchor::try_from_cert_der(&der.0)
+            .map_err(|e| Error::InvalidCertificateData(format!("{}", e)))?;
```

Comment:
I suggest we use a different, new, error variant: `InvalidRootCertData`. `InvalidCertificateData` should only be returned during certificate validation.

Reply:
i can understand that though
based on the description and error cases in the "try_from_cert_der" function, an error variant called "FailtedToDecodeDER" seems more appropriate, as the function doesn't actually validate the certificate, only decodes it

Reply:
> i can understand that though
> based on the description and error cases in the "try_from_cert_der" function, an error variant called "FailtedToDecodeDER" seems more appropriate, as the function doesn't actually validate the certificate, only decodes it

The problem I see with `FailedtoDecodeDER` is that it doesn't provide any context to the user about what was trying to be decoded. Also, I think `InvalidRootCertData` could be used in the future for any kind of problem with the root cert, not just problems with the ASN.1 encoding.

Reply:
You are right, `FailedToDecodeDER` by name lacks context, in which the error occurred. I don't like that `InvalidRootCertData` suggests something (a validation of the root certificate) which doesn't actually happen.
What do you think about naming the error `FailedToDecodeRootCertData`?